bootstrap.py

```diff
@@ -369,13 +370,29 @@ def return_puppetenv_for_hg(hg_id):
     else:
         return 'production'
 
+def create_domain(domain, orgid, locid):
+    myurl = "https://" + options.foreman_fqdn + ":" + API_PORT + "/api/v2/domains"
+    domid = return_matching_foreman_key('domains', 'name="%s"' % domain, 'id', True)
+    if not domid:
+        jsondata = json.loads('{"domain": {"name": "%s", "organization_ids": [%s], "location_ids": [%s]}}' % (domain, orgid, locid))
+        print_running("Calling Foreman API to create domain %s associated with the org & location" % domain)
+        post_json(myurl, jsondata)
+        return domain
+    else:
+        return domain
 
 def create_host():
     myhgid = return_matching_foreman_key('hostgroups', 'title="%s"' % options.hostgroup, 'id', False)
     mylocid = return_matching_foreman_key('locations', 'title="%s"' % options.location, 'id', False)
     myorgid = return_matching_foreman_key('organizations', 'name="%s"' % options.org, 'id', False)
     if DOMAIN:
-        mydomainid = return_matching_foreman_key('domains', 'name="%s"' % DOMAIN, 'id', False)
+        if options.add_domain:
+            create_domain(DOMAIN, myorgid, mylocid)
+        try:
```

Comment:
I think I'd prefer:

    mydomainid = return_matching_foreman_key('domains', 'name="%s"' % DOMAIN, 'id', True)
    if not mydomainid:
        print_generic("Domain %s doesn't exist in Foreman, consider using the --add-domain option." % DOMAIN)
        sys.exit(2)